src/seedu/addressbook/commands/SortCommand.java

```diff
@@ -0,0 +1,114 @@
+package seedu.addressbook.commands;
+
+import seedu.addressbook.data.person.Person;
+import seedu.addressbook.common.Messages;
+import seedu.addressbook.data.person.ReadOnlyPerson;
+import seedu.addressbook.data.person.UniquePersonList;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+/**
+ * Sorts the address book based on the input order.
+ */
+public class SortCommand extends Command {
+
+    public static final String COMMAND_WORD = "sort";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Sorts the address book by name based on the input order.\n"
+            + "Parameters: ascend, descend\n"
+            + "Example: " + COMMAND_WORD + " ascend";
+
+    private final String commandArgs;
+
+    public SortCommand(String commandArgs) {
+        this.commandArgs = commandArgs;
+    }
+
+    @Override
+    public CommandResult execute() {
+        if (executeSortList(commandArgs)) {
+            List<ReadOnlyPerson> allPersons = addressBook.getAllPersons().immutableListView();
+            return new CommandResult(Messages.MESSAGE_PERSONS_SORTED_OVERVIEW, allPersons);
+        } else {
+            return new CommandResult(String.format(Messages.MESSAGE_INVALID_COMMAND_FORMAT, MESSAGE_USAGE));
```

Comment:
Noticed that the invalid command format message applies to a case of duplicate persons as well. Think about how this could be improved.